app.js

```diff
@@ -800,18 +816,19 @@ function handleSignal(signal) {
 
 }
 
-// Convert Uppercase '_' delimited string to TitleCase replacing _ with spaces
+// Convert '_' delimited string to Title Case, preserving existing title case and replacing '_' with spaces
 function toTitleCase(str) {
```

Comment:
Please give me an example (from my parameters) where you think the original code is bugged.  I KNOW that all the OpenThings parameters are all uppercase so there is really no need to convert the first character.

It could be made even more efficient by copying the first character before converting the whole string toLowerCase.

Reply:
The first iteration of the function that you provided didn't  capitalise on the first letter of words that are only one word long.

Reply:
It doesn't need to it is always in uppercase

Reply:
`switch` is lowercase in `devices/2.json`
`retries` `command``last_seen` `battery` are lowercase in `devices/3.json`

The amended code fixes this and title cases them :)

Reply:
![image](https://github.com/Achronite/mqtt-energenie-ener314rt/assets/5207490/617e7202-55bf-4843-8419-6d3f46df19a9)

vs
![image](https://github.com/Achronite/mqtt-energenie-ener314rt/assets/5207490/776e1420-393c-4127-9b81-fd3b4ca27663)